src/main/java/data/FileDatabase.java

```diff
@@ -0,0 +1,43 @@
+package data;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+/* This class creates an instance of a database that a file can be written to and read from.
+Assumptions: for each runID, only 1 of each datatype can be associated with it. */
+public class FileDatabase implements Database {
+  /* Folder for Database. */
+  File folder;
+
+  /* Creates instance of a database with this name. */
+  public FileDatabase(String name) throws IOException {
+    folder = new File(name);
+    folder.mkdir();
```

Comment:
Check for failure here. `File.mkdir` can return false, which would mean that all the other operations on the database are invalid.

https://docs.oracle.com/javase/8/docs/api/java/io/File.html#mkdir--

Reply:
thanks! i added a try catch to resolve. i  thought by adding the IOExeption of the line of the method signature that would check for failure. should i take this out now that i have the try catch?

Reply:
(Note: I don't see any new commits since my review, so I can't see the change you're referring to.)

Adding `throws IOException` to the method signature only signifies that it's _allowed_ for the method to throw that exception. It's mostly a form of documentation for the callers, so that they know the method can fail. It doesn't actually change the behavior of the function at all, and it's possible to have a method marked `throws IOException` that is actually incapable of throwing. That's what you have here.

The compiler _does_ enforce the converse: if a method can throw an exception, the method signature must include the `throws` clause. Consider these two methods.
```
private int f() {
  return 0;
}
private int g() throws IOException {
  return 0;
}
```
It's not possible for either method to throw an exception, because nothing in either method body would cause it. The two methods behave identically. The difference is in how callers have to behave.
```
private int h1() {
  return f(); // This is fine.
}
private int h2() {
  return g(); // Compilation error. g() can throw, but h2() can't. The exception has nowhere to go.
}
private int h3() {
  try {
    return g(); // This is fine. We catch the potential exception, and h3() doesn't throw anything.
  } catch (IOException e) {
    return -1;
  }
}
private int h4() throws IOException {
  return g(); // This is fine. We don't catch the exception g() throws, so h4() will pass the exception along to its caller, which it's allowed to do.
}
```

For more detail, see https://docs.oracle.com/javase/tutorial/essential/exceptions/index.html.

Reply:
Now, back to this specific case: `File.mkdir` doesn't throw `IOException` on failure, it reports failure via the return value. It's up to you to check that return value and handle the error how you want.